client/ayon_core/addon/interfaces.py

```diff
@@ -265,36 +304,40 @@ def tray_menu(self, tray_menu):
         action.triggered.connect(self.on_action_trigger)
         self._action_item = action
 
-    def tray_start(self):
+    def tray_start(self) -> None:
+        """Start procedure in tray tool."""
         return
 
-    def tray_exit(self):
+    def tray_exit(self) -> None:
+        """Cleanup method which is executed on tray shutdown."""
         return
 
 
 class ITrayService(ITrayAddon):
+    """Tray service Interface."""
     # Module's property
-    menu_action = None
+    menu_action: QtWidgets.QAction = None
```

Comment:
How did not the `QtWidgets` usages crash for you?

Reply:
This works because it is imported only in `TYPE_CHECKING`

Reply:
Ok, `from __future__ import annotations` seems like magical replacer...